### PR TITLE
MM-26166 incident metadata endpoint

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
This adopts the proposal from the REST API to call the "details" endpoint "metadata", and bury it in all the "computed" things about an incident. This now only returns that metadata, and not incident + metadata, simplifying the naming somewhat.

As part of this change, and as discussed on -daily, I'm returning checklists on both `/incidents` and `/incidents/<id>` to unify the API.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-26166

### CODE REVIEW

File: webapp/src/components/backstage/incidents/incident_details/incident_details.tsx [hunk 1]
```diff
@@ -241,36 +241,36 @@ const FetchingStateType = {
...
-        fetchIncident(match.params.incidentId);
+        const incidentId = match.params.incidentId;
+
+        Promise.all([fetchIncident(incidentId), fetchIncidentMetadata(incidentId)]).then(([incidentResult, incidentMetadataResult]) => {
+            setIncident(incidentResult);
+            setIncidentMetadata(incidentMetadataResult);
+            setFetchingState(FetchingStateType.fetched);
+        }).catch(() => {
+            setFetchingState(FetchingStateType.notFound);
+        });
```
Comment (agarciamontoro) [inline webapp/src/components/backstage/incidents/incident_details/incident_details.tsx:261]: Nice!

**Review by agarciamontoro — APPROVED** [review]
Much much cleaner, thank you! :tada:

File: server/incident/incident.go [hunk 2]
```diff
@@ -88,6 +82,30 @@ type GetIncidentsResults struct {
...
+	newGetIncidentsResults.Items = make([]Incident, 0, len(r.Items))
+	for _, i := range r.Items {
+		newGetIncidentsResults.Items = append(newGetIncidentsResults.Items, *i.Clone())
+	}
+
+	return newGetIncidentsResults
+}
+
+func (r GetIncidentsResults) MarshalJSON() ([]byte, error) {
+	type Alias GetIncidentsResults
```
[the remark this answers is not in the document]
Comment (lieut-data) [reply server/incident/incident.go:97]: Aliasing the type allows us to use `json.Marshal` without recursing into ourselves here.

[the remark this answers is not in the document]
Comment (lieut-data) [reply server/incident/incident.go:97]: (Since `MarshalJSON` is not defined /on/ `Alias`)

[the remark this answers is not in the document]
Comment (agarciamontoro) [reply server/incident/incident.go:97]: Aaaah, ok, I see. Thanks for the clarification!

**Review by cpoile — APPROVED** [review]

**Pull request merged**